src/__tests__/search-and-replace-test.jsx

```diff
@@ -0,0 +1,185 @@
+var TestUtils = React.addons.TestUtils;
+
+var itemJson = require("./test-items/search-and-replace-item.json");
+var particleJson = require("./test-items/search-and-replace-particle.json");
+
+describe("Search and Replace", function() {
+    var editor, editorProps;
+    var searchInput, replaceInput;
+    var previousButton, nextButton, replaceButton, replaceAllButton;
+
+    beforeEach(function () {
+        var problemNum = Math.floor(Math.random() * 100);
+        var enabledFeatures = {
+            highlight: true,
+            toolTipFormats: true,
+            useMathQuill: true
+        };
+
+        editorProps = {
+            problemNum: problemNum,
+            enabledFeatures: enabledFeatures,
+            developerMode: true,
+            imageUploader: function(image, callback) {
+                _.delay(callback, 1000, "http://fake.image.url");
+            },
+            apiOptions: {
+                fancyDropdowns: true,
+                __onInputError: function() {
+                    var args = _.toArray(arguments);
+                    console.log.apply(console, ["onInputError:"].concat(args));
+                    return true;
+                },
+                __interceptInputFocus: function() {
+                    var args = _.toArray(arguments);
+                    console.log.apply(console, ["interceptInputFocus:"].concat(args));
+                    return;
+                },
+                onFocusChange: function(newPath, oldPath) {
+                    console.log("onFocusChange", newPath, oldPath);
+                },
+                __staticRender: true
+            }
+        };
+    });
+
+    describe("EditorPage", function() {
+        beforeEach(function() {
+            editorProps = $.extend(true, editorProps, itemJson);
+            editorProps.componentClass = EditorPage;
+
+            var StatefulEditorPageFactory = React.createFactory(StatefulEditorPage);
+            editor = React.render(
+                StatefulEditorPageFactory(editorProps, null),
+                document.body
+            );
+
+            var dialog = editor.refs.searchAndReplace;
+
+            searchInput = dialog.refs.searchInput.getDOMNode();
+            replaceInput = dialog.refs.replaceInput.getDOMNode();
+            replaceButton = dialog.refs.replaceButton.getDOMNode();
+            replaceAllButton = dialog.refs.replaceAllButton.getDOMNode();
+            previousButton = dialog.refs.previousButton.getDOMNode();
+            nextButton = dialog.refs.nextButton.getDOMNode();
+        });
+
+        afterEach(function (done) {
+            React.unmountComponentAtNode(document.body);
+            document.body.innerHTML = "";
+            setTimeout(done);
+        });
+
+        it("should replace all instances of the searchString not inside a widget reference", function() {
+            React.addons.TestUtils.Simulate.change(searchInput, {target: {value: "categorizer"}});
```

Comment:
I'll fix this.  I had some troubles running ```runlint.py```.  There are node modules which it depends on that need to be installed.  If the last line of a ```.jsx``` file is empty but jshint reports and error on it an exception will occur because python doesn't include that empty string in the array produces by ```readlines```.  I'll put together a pull request for khan-linter tomorrow to fix these issues.  It's catching the long lines which is good, but it's also reporting hundreds of other violations in react-components and other sub-repos.  Is this normal?